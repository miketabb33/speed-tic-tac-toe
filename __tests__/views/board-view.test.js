import React from 'react'
import renderer from 'react-test-renderer'
import BoardView from '../../views/board-view'
import MockBoard from '../../mocks/mock-board'

const whateverIndex = 1
const mockBoard = new MockBoard

test('render', () => {
	const subject = renderer.create(
		<BoardView 
			squares = { mockBoard.getSquares(whateverIndex, ['X']) }
			boardIndex = { whateverIndex }
		/>
	)
	
	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})