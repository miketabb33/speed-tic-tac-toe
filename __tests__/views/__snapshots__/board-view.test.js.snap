// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="board"
>
  <div
    className="boardImage"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzMwIiBoZWlnaHQ9IjMzMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Fboard.png&w=750&q=75"
          srcSet="/_next/image?url=%2Fboard.png&w=384&q=75 1x, /_next/image?url=%2Fboard.png&w=750&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
  </div>
  <div
    className="boardSquares"
  >
    <div>
      <button
        className="square"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzkiIGhlaWdodD0iNzkiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <img
            alt="X player marker"
            data-nimg="intrinsic"
            decoding="async"
            draggable="false"
            src="/_next/image?url=%2Fplayer-markers%2FX.png&w=256&q=75"
            srcSet="/_next/image?url=%2Fplayer-markers%2FX.png&w=96&q=75 1x, /_next/image?url=%2Fplayer-markers%2FX.png&w=256&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="X player marker"
              data-nimg="intrinsic"
              decoding="async"
              draggable="false"
              loading="lazy"
              src="/_next/image?url=%2Fplayer-markers%2FX.png&w=256&q=75"
              srcSet="/_next/image?url=%2Fplayer-markers%2FX.png&w=96&q=75 1x, /_next/image?url=%2Fplayer-markers%2FX.png&w=256&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </div>
      </button>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
    </div>
    <div>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
    </div>
    <div>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
      <button
        className="square"
        onClick={[Function]}
      >
        <svg
          height={79}
          width={79}
        >
          <rect
            className="clear"
            height={79}
            width={79}
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="boardTitle"
  />
</div>
`;
